client_integration_test.go

```diff
@@ -651,6 +651,69 @@ func TestClientRemove(t *testing.T) {
 	}
 }
 
+func TestClientRemoveAll(t *testing.T) {
+	sftp, cmd := testClient(t, READWRITE, NODELAY)
+	defer cmd.Wait()
+	defer sftp.Close()
+
+	// Create a temporary directory for testing
+	tempDir, err := ioutil.TempDir("", "sftptest-removeAll")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	// Create a directory tree
+	dir1, err := ioutil.TempDir(tempDir, "foo")
+	if err != nil {
+		t.Fatal(err)
+	}
+	dir2, err := ioutil.TempDir(dir1, "bar")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create some files within the directory tree
+	file1 := tempDir + "/file1.txt"
+	file2 := dir1 + "/file2.txt"
+	file3 := dir2 + "/file3.txt"
+	err = ioutil.WriteFile(file1, []byte("File 1"), 0644)
+	if err != nil {
+		t.Fatalf("Failed to create file: %v", err)
+	}
+	err = ioutil.WriteFile(file2, []byte("File 2"), 0644)
+	if err != nil {
+		t.Fatalf("Failed to create file: %v", err)
+	}
+	err = ioutil.WriteFile(file3, []byte("File 3"), 0644)
+	if err != nil {
+		t.Fatalf("Failed to create file: %v", err)
```

Comment:
[fyi] `t.Fatalf("…: %v", err)` is equivalent to `t.Fatal("…:", err)`

Reply:
Okay!

Reply:
No, this PR still has value. It does exist in `os.RemoveAll`.

But `os.Remove´ and `Client.Remove` only work on _empty_ directories. Unlike `os.RemoveAll`. If `Client.Remove` was not working on empty directories, then that’s a problem.

Reply:
Okay! I just rechecked again and client.Remove() is deleting empty directory. Its weird earlier I was getting error but when I tested it today it deleted empty directory. (I cleared cookie and then retry and it worked now)

Reply:
It might have had something in it when you tried before? Especially like a hidden file or something? 🤷‍♀️

Reply:
No I just had multiple files for testing, all the files deleted but empty directory was not deleted earlier. But today when I test it again and debug it, first all files got deleted and then empty directory.